recipes-extended/realsense-camera/realsense-camera/0001-librealsense-Fix-link-when-the-system-library-is-use.patch

```diff
@@ -0,0 +1,30 @@
+From fb74e5eec2368f20c4b43cf9224519b29aee35ad Mon Sep 17 00:00:00 2001
+From: Murilo Belluzzo <murilo.belluzzo@intel.com>
+Date: Wed, 5 Apr 2017 17:06:44 -0700
+Subject: [PATCH 3/3] librealsense: Fix link when the system library is used
+
+Otherwise librealsense won't be listed as a dependency for the nodelet
+that will fail to load due to undefined symbols.
+
+Signed-off-by: Murilo Belluzzo <murilo.belluzzo@intel.com>
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index a1ec2f6..926992a 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -106,6 +106,12 @@ add_library(${PROJECT_NAME}_nodelet src/base_nodelet.cpp src/r200_nodelet.cpp sr
+ target_link_libraries(${PROJECT_NAME}_nodelet
+   ${catkin_LIBRARIES}
+ )
++if(USE_SYSTEM_LIBREALSENSE)
++  target_link_libraries(${PROJECT_NAME}_nodelet
++    ${realsense_LIBRARY}
```

Comment:
The problem of using "realsense::realsense" directly is that only works with master for now (there is a lot of custom Findrealsense.cmake out there).

Also is a good cmake practice to use the variables that find_package() is supposed to set instead of explicitly state the supposed values of those. Maintain consistency with other uses and is more future proof.

Another pro is that keep this patch exactly the same of what was merged upstream.